### PR TITLE
Add celu, hardsigmoid, selu, thresholdedrelu layers

### PR DESCRIPTION
**Pull request by rogday**
Tests are coming to 4.x with #21088

ONNX reference:
[Celu](https://github.com/onnx/onnx/blob/master/docs/Operators.md#celu), [HardSigmoid](https://github.com/onnx/onnx/blob/master/docs/Operators.md#hardsigmoid), [Selu](https://github.com/onnx/onnx/blob/master/docs/Operators.md#selu), [ThresholdedReLU](https://github.com/onnx/onnx/blob/master/docs/Operators.md#thresholdedrelu)

Note:
I removed `initWebnnAPI` blocks from layers, which already derive desired functionality from `BaseDefaultFunctor`. (intoduced in https://github.com/opencv/opencv/pull/20406)

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [X] The PR is proposed to proper branch
- [X] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac

buildworker:Custom=linux-4,linux-6
Xbuild_image:Custom=ubuntu-cuda:16.04
build_image:Custom=ubuntu-cuda:18.04

Xbuild_image:Custom=ubuntu-openvino-2021.4.2:20.04
Xbuild_image:Custom=ubuntu-openvino-2021.3.0:20.04
Xbuild_image:Custom=ubuntu-openvino-2021.2.0:20.04
Xbuild_image:Custom=ubuntu-openvino-2020.3.0:16.04
build_image:Custom Win=openvino-2021.4.2
Xbuild_image:Custom Win=openvino-2021.3.0-vc16
build_image:Custom Mac=openvino-2021.4.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

Xbuildworker:Custom=linux-1
Xtest_opencl:Custom=ON
Xtest_bigdata:Custom=1
Xtest_filter:Custom=*

test_bigdata:Custom Win=1
buildworker:Custom Win=windows-3
test_opencl:Custom Win=ON
test_filter:Custom Win=*
```

### CODE REVIEW

File: modules/dnn/src/cuda/functors.hpp [hunk 1]
```diff
@@ -299,6 +299,84 @@ struct NotFunctor {
...
+    };
+
+    CUDA4DNN_DEVICE CeluFunctor() : CeluFunctor(Params{}) { }
+    CUDA4DNN_DEVICE CeluFunctor(const Params& params) : alpha{params.alpha} { }
+
+    CUDA4DNN_DEVICE T operator()(T value) {
+        using csl::device::min;
+        using csl::device::max;
+        using csl::device::exp;
+        return max(T(0), value) + min(T(0), alpha * (exp(value / alpha) - T(1)));
```
Comment (YashasSamaga) [inline modules/dnn/src/cuda/functors.hpp:317]: `expm1`?

File: modules/dnn/src/cuda/functors.hpp [hunk 2]
```diff
@@ -299,6 +299,84 @@ struct NotFunctor {
...
+        CUDA4DNN_HOST_DEVICE Params(T alpha_, T gamma_) : alpha(alpha_), gamma(gamma_) { }
+        T alpha, gamma;
+    };
+
+    CUDA4DNN_DEVICE SeluFunctor() : SeluFunctor(Params{}) { }
+    CUDA4DNN_DEVICE SeluFunctor(const Params& params): alpha{params.alpha}, gamma{params.gamma} { }
+
+    CUDA4DNN_DEVICE T operator()(T value) {
+        using csl::device::exp;
+        return gamma * ((value > T(0)) ? value : (alpha * exp(value) - alpha));
```
Comment (YashasSamaga) [inline modules/dnn/src/cuda/functors.hpp:356]: 1. `alpha * expm1(value)`?
2. Were the double parenthesis for `value > T(0)` intended?

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Comment by alalek** [comment]
Need to rebase PR.

**Comment by Nokalm** [comment]
If I'm not wrong these layers are merged in opencv 4.7.0 ? Are they implemented only for CUDA backends ? or are they available on default CPU ? Because when I try to load my rtmdet.onnx not far away from a yolo I get the error below:

> Can't create layer "/backbone/stage1/stage1.1/attention/act/HardSigmoid_output_0" of type "HardSigmoid" in function 'getLayerInstance'

Thank you

**Pull request merged**